landoapi/api/revisions.py

```diff
@@ -26,6 +27,28 @@ def get(revision_id, api_key=None):
     return _format_revision(phab, revision, include_parents=True), 200
 
 
+def land(revision_id, api_key=None):
+    """ API endpoint at /revisions/{id}/transplants to land revision. """
+    phab = PhabricatorClient(api_key)
+    revision = phab.get_revision(id=revision_id)
+    revision = _format_revision(phab, revision, include_parents=True)
+    if not revision:
+        # We could not find a matching revision.
+        return problem(
+            404,
+            'Revision not found',
+            'The requested revision does not exist',
+            type='https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404'
+        )
+
+    trans = TransplantClient()
+    id = trans.land(
+        'ldap_username@example.com', revision['repo']['url'], 'patch',
+        'destination', 'push_bookmark', 'http://pingback.url'
+    )
+    return {'request_id': id}, 202
```

Comment:
Ok, here is a big comment - let me know if I am just misunderstanding something.

I think this is a flaw in how the trello card was designed. Returning a transplant request id will serve no purpose in letting the UI or any other client know the status of the land request. The UI has no database and no way to store state - and so if the UI ever asks to know whether a revision has been landed, it will ask using the revision ID which comes from the URL of the page the user loads (e.g. lando.com/D1234).

The request ID is important for lando-api to have of course, since it can store state and track all the landing statuses internally. Lando-api can save that request_id to its database and later use it to look up its status. But, returning this info via the API won't help anyone, since no one other than lando can talk to the transplant service. Is this API just a intermediate step for to get the card done?

I propose that this endpoint just return a 200 status code if it succeeded to be added to the landing queue, or a bad status code if transplant failed immediately. Then later there will be another endpoint that accept a revision id to see whether that revision has been actually been landed.

Taking this into account the `/revisions/{id}/transplants` route doesn't really make as much sense. Routes such as `/revisions/{id}/land` and `/revisions/{id}/landing_status` would make more sense imo.

cc @mars-f